src/sentry/api/endpoints/project_group_index.py

```diff
@@ -346,74 +376,112 @@ def put(self, request, project):
+                groups_with_releases = set(releases_by_group.iterkeys())
+                # Determine which groups DO NOT have releases
+                to_resolve_normally = group_ids - groups_with_releases
+            else:
+                groups_with_releases = None
+                to_resolve_normally = group_ids
 
-        elif result.get('status') == 'resolved':
             now = timezone.now()
 
-            happened = queryset.exclude(
-                status=GroupStatus.RESOLVED,
-            ).update(
-                status=GroupStatus.RESOLVED,
-                resolved_at=now,
-            )
+            # First, handle the groups that DO NOT have release data
+            if to_resolve_normally:
+                happened = Group.objects.filter(
+                    id__in=to_resolve_normally,
+                ).exclude(
+                    status=GroupStatus.RESOLVED,
+                ).update(
+                    status=GroupStatus.RESOLVED,
+                    resolved_at=now,
+                )
 
-            GroupResolution.objects.filter(
-                group__in=group_ids,
-            ).delete()
+                GroupResolution.objects.filter(
+                    group__in=to_resolve_normally,
+                ).delete()
 
-            if group_list and happened:
-                for group in group_list:
-                    group.status = GroupStatus.RESOLVED
-                    group.resolved_at = now
-                    activity = Activity.objects.create(
-                        project=group.project,
-                        group=group,
-                        type=Activity.SET_RESOLVED,
-                        user=acting_user,
+                if happened:
+                    for group in group_list:
+                        # ignore the groups that we're not resolving
+                        if group.id not in to_resolve_normally:
+                            continue
+                        group.status = GroupStatus.RESOLVED
+                        group.resolved_at = now
+                        activity = Activity.objects.create(
+                            project=group.project,
+                            group=group,
+                            type=Activity.SET_RESOLVED,
+                            user=acting_user,
+                        )
+                        activity.send_notification()
+
+            # Next, handle groups that DO have release data
+            if groups_with_releases:
+                Group.objects.filter(
+                    id__in=groups_with_releases,
+                ).update(
+                    status=GroupStatus.RESOLVED,
+                    resolved_at=now,
+                )
+
+                for release in releases:
+                    self._resolve_in_release(
+                        actor=acting_user,
+                        group_list=[g for g in group_list if g.id in groups_with_releases],
```

Comment:
This isn't handling the case where there are more than 1 release. Going to write a test for this as well.